### DISCUSSION TITLE
CSV Importer: Has no Date Role in the Role Configuration drop down

### DISCUSSION THREAD

**Discussion by navin2004** [firefly-iii/firefly-iii#8441] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I see issues on this closed, but I am a little lost here. /import/roles page does not have any Role field called Date. I have a transaction date in my CSV that I downloaded from Mint. I have so far tried 10 times and all I see for Date to map against is Transaction Date. Documentation mentions a Data field, but I don't see that at all.
I am using the latest and greatest version of Firefly on Docker.
![RoleDefinition Dropdown has no date](https://github.com/firefly-iii/firefly-iii/assets/25390585/77384900-45f5-4a1d-9efa-bfb131e57274)
Even if documentation is not updated and this Data is now called Transaction Date. It does not work, all my transactions have the day of import as the date.
I have the data set to format n-d-Y as my month is single digit. Like this : 6/23/2023

### Debug information

Debug information generated at 2024-01-21 18:14:17 America/New_York for Firefly III Data Importer version **1.4.1**.

<table><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>Version</td><td>1.4.1</td></tr><tr><td>Build</td><td><span>#</span>, base <span>#</span>78</td></tr><tr><td>System</td><td>PHP 8.3.1, Linux, apache2handler</td></tr></table><table><tr><th colspan="2">App information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>Timezone</td><td>America/New_York, America/Toronto</td></tr><tr><td>Environment</td><td>local</td></tr><tr><td>Debug mode</td><td>false, cache 'file'</td></tr><tr><td>Log level</td><td>debug, stack</td></tr><tr><td>Display errors</td><td>Off, ALL errors</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Trusted proxies</td><td></td></tr></table><table><tr><th colspan="2">User information</th></tr><tr><th>Item</th><th>Value</th></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36 Edg/120.0.0.0</td></tr></table>

### Expected behaviour

I expect CSV Importer role of Transaction Date to match to Date in the Asset Account

**Comment by JC5** [comment]
https://docs.firefly-iii.org/references/faq/data-importer/import/#all-transactions-are-imported-as-today

**Reply by navin2004** [reply]
I read that doc and doing things as it says. I just re-tried by changing the header name from Date to Transaction Date and it still did not work. Just puts current date.
Date in CSV is "1/11/2024" or "12/19/2022".
So I used the format n-d-Y after reading the documentation.
I just have 107 rows in this file and headers do not repeat. I do not get any failures. Just a warnings, which does not impact currency. "Line #0: Line #0 may have had its currency changed (from ID #1 to ID #10). This happens because the associated asset account overrules the currency of the transaction."

**Reply by JC5** [reply, marked as answer]
If your dates are "1/11/2024" or "12/19/2022", you should use format `n/d/Y`.

**Reply by navin2004** [reply]
I feel so stupid now missing that....thank you.
